src/screens/ExchangeModal.tsx

```diff
@@ -436,27 +445,31 @@ export default function ExchangeModal({
 
   const updateGasLimit = useCallback(async () => {
     try {
-      const swapParams = {
+      const provider = await getProviderForNetwork(currentNetwork);
+      const swapParams:
+        | SwapActionParameters
+        | CrosschainSwapActionParameters = {
         chainId,
         inputAmount: inputAmount!,
         outputAmount: outputAmount!,
-        provider: currentProvider!,
+        provider,
         tradeDetails: tradeDetails!,
       };
 
-      const rapType = getSwapRapTypeByExchangeType(type);
+      const rapType = getSwapRapTypeByExchangeType(type, isCrosschainSwap);
       const gasLimit = await getSwapRapEstimationByType(rapType, swapParams);
       if (gasLimit) {
         if (currentNetwork === Network.optimism) {
           if (tradeDetails) {
             const l1GasFeeOptimism = await ethereumUtils.calculateL1FeeOptimism(
+              // @ts-ignore
```

Comment:
I don't think this ignore statement is doing anything.

Reply:
right!